// Copyright 2018 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Configuration information for processing an external segment database.

syntax = "proto2";

import "fonbund/segment_normalizer_config.proto";

package fonbund;

// This message represents information about individual phonological segment
// repositories.
//
// Next available ID: 13
message SegmentRepositoryConfig {
  // Name of the repository.
  optional string name = 1;

  // Field separator used for parsing the segment repository.
  optional string field_separator = 2;

  // The first entry in the repository contains the description of all the
  // columns.
  optional bool has_column_description = 3;

  // The (0-based) offset of the column containing the segment names. All the
  // columns up to this column are ignored.
  optional int32 segment_column_id = 4;

  // The (0-based) IDs of the original columns that should be ignored. This list
  // should not include the <segment_column_id> column ID defined above.
  repeated int32 ignore_column_ids = 5;

  // Expected number of distinctive features defined in the repository. This
  // number has to be updated every time the repository structure changes.
  // This number is used solely for sanity checks upon the repository loading.
  optional int32 num_features = 6;

  // If enabled, the segments in the repository are assumed to be represented by
  // multivalued features.
  optional bool multivalued_features = 7;

  // Default value for multivalued feature: Use when the value is missing.
  optional string multivalued_feature_default_value = 8;

  // Configuration for normalizing the segments.
  optional SegmentNormalizerConfig normalizer = 9;

  // If enabled, a list of languages represented by <language_regions>
  // represents languages whitelisted for processing. Otherwise, that list
  // represents a blacklist containing languages that shouldn't be processed.
  optional bool language_list_whitelisted = 10;

  // List of languages which are either black-listed or white-listed for
  // processing depending on the value of
  repeated string language_regions = 11;

  // Extra segment repository lines representing a verbatim string encoding of
  // any extra segment, feature values that for some reason we don't want to
  // keep as part of the main segment repository. Please note, the format of
  // each string has to match the format of entries in the main repository.
  //
  // Please note: This field is unused at the moment because we opted for
  // modifying the third-party inventories in-place. It will become very useful
  // once we change this policy.
  repeated string extra_segment_feature_entries = 12;
}
